File: src/.NET/fix40/AllocationACK.cs
namespace QuickFix40
{

  public class AllocationACK : Message
  {
    public AllocationACK() : base(MsgType()) {}
    static QuickFix.MsgType MsgType() { return new QuickFix.MsgType("P"); }

    public AllocationACK(
      QuickFix.AllocID aAllocID,
      QuickFix.TradeDate aTradeDate,
      QuickFix.AllocStatus aAllocStatus )
    : base(MsgType()) {
      set(aAllocID);
      set(aTradeDate);
      set(aAllocStatus);
    }

    public void set(QuickFix.ClientID value)
    { setField(value); }
    public QuickFix.ClientID get(QuickFix.ClientID  value)
    { getField(value); return value; }
    public QuickFix.ClientID getClientID()
    { QuickFix.ClientID value = new QuickFix.ClientID();
      getField(value); return value; }
    public bool isSet(QuickFix.ClientID field)
    { return isSetField(field); }
    public bool isSetClientID()
    { return isSetField(109); }

    public void set(QuickFix.ExecBroker value)
    { setField(value); }
    public QuickFix.ExecBroker get(QuickFix.ExecBroker  value)
    { getField(value); return value; }
    public QuickFix.ExecBroker getExecBroker()
    { QuickFix.ExecBroker value = new QuickFix.ExecBroker();
      getField(value); return value; }
    public bool isSet(QuickFix.ExecBroker field)
    { return isSetField(field); }
    public bool isSetExecBroker()
    { return isSetField(76); }

    public void set(QuickFix.AllocID value)
    { setField(value); }
    public QuickFix.AllocID get(QuickFix.AllocID  value)
    { getField(value); return value; }
    public QuickFix.AllocID getAllocID()
    { QuickFix.AllocID value = new QuickFix.AllocID();
      getField(value); return value; }
    public bool isSet(QuickFix.AllocID field)
    { return isSetField(field); }
    public bool isSetAllocID()
    { return isSetField(70); }

    public void set(QuickFix.TradeDate value)
    { setField(value); }
    public QuickFix.TradeDate get(QuickFix.TradeDate  value)
    { getField(value); return value; }
    public QuickFix.TradeDate getTradeDate()
    { QuickFix.TradeDate value = new QuickFix.TradeDate();
      getField(value); return value; }
    public bool isSet(QuickFix.TradeDate field)
    { return isSetField(field); }
    public bool isSetTradeDate()
    { return isSetField(75); }

    public void set(QuickFix.TransactTime value)
    { setField(value); }
    public QuickFix.TransactTime get(QuickFix.TransactTime  value)
    { getField(value); return value; }
    public QuickFix.TransactTime getTransactTime()
    { QuickFix.TransactTime value = new QuickFix.TransactTime();
      getField(value); return value; }
    public bool isSet(QuickFix.TransactTime field)
    { return isSetField(field); }
    public bool isSetTransactTime()
    { return isSetField(60); }

    public void set(QuickFix.AllocStatus value)
    { setField(value); }
    public QuickFix.AllocStatus get(QuickFix.AllocStatus  value)
    { getField(value); return value; }
    public QuickFix.AllocStatus getAllocStatus()
    { QuickFix.AllocStatus value = new QuickFix.AllocStatus();
      getField(value); return value; }
    public bool isSet(QuickFix.AllocStatus field)
    { return isSetField(field); }
    public bool isSetAllocStatus()
    { return isSetField(87); }

    public void set(QuickFix.AllocRejCode value)
    { setField(value); }
    public QuickFix.AllocRejCode get(QuickFix.AllocRejCode  value)
    { getField(value); return value; }
    public QuickFix.AllocRejCode getAllocRejCode()
    { QuickFix.AllocRejCode value = new QuickFix.AllocRejCode();
      getField(value); return value; }
    public bool isSet(QuickFix.AllocRejCode field)
    { return isSetField(field); }
    public bool isSetAllocRejCode()
    { return isSetField(88); }

    public void set(QuickFix.Text value)
    { setField(value); }
    public QuickFix.Text get(QuickFix.Text  value)
    { getField(value); return value; }
    public QuickFix.Text getText()
    { QuickFix.Text value = new QuickFix.Text();
      getField(value); return value; }
    public bool isSet(QuickFix.Text field)
    { return isSetField(field); }
    public bool isSetText()
    { return isSetField(58); }

  };

}

File: src/java/src/org/quickfix/fix40/AllocationACK.java
package org.quickfix.fix40; 
import org.quickfix.FieldNotFound; 
import org.quickfix.Group; 
import org.quickfix.field.*; 

public class AllocationACK extends Message 
{ 

  public AllocationACK() 
  { 
    getHeader().setField(new MsgType("P")); 
  } 
  public AllocationACK(    
    org.quickfix.field.AllocID aAllocID,    
    org.quickfix.field.TradeDate aTradeDate,    
    org.quickfix.field.AllocStatus aAllocStatus ) 
  {  
    getHeader().setField(new MsgType("P")); 
    set(aAllocID); 
    set(aTradeDate); 
    set(aAllocStatus);  
  } 

  public void set(org.quickfix.field.ClientID value) 
  { setField(value); } 
  public org.quickfix.field.ClientID get(org.quickfix.field.ClientID value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.ClientID getClientID() throws FieldNotFound 
  { org.quickfix.field.ClientID value = new org.quickfix.field.ClientID(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.ClientID field) 
  { return isSetField(field); } 
  public boolean isSetClientID() 
  { return isSetField(109); } 

  public void set(org.quickfix.field.ExecBroker value) 
  { setField(value); } 
  public org.quickfix.field.ExecBroker get(org.quickfix.field.ExecBroker value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.ExecBroker getExecBroker() throws FieldNotFound 
  { org.quickfix.field.ExecBroker value = new org.quickfix.field.ExecBroker(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.ExecBroker field) 
  { return isSetField(field); } 
  public boolean isSetExecBroker() 
  { return isSetField(76); } 

  public void set(org.quickfix.field.AllocID value) 
  { setField(value); } 
  public org.quickfix.field.AllocID get(org.quickfix.field.AllocID value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AllocID getAllocID() throws FieldNotFound 
  { org.quickfix.field.AllocID value = new org.quickfix.field.AllocID(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AllocID field) 
  { return isSetField(field); } 
  public boolean isSetAllocID() 
  { return isSetField(70); } 

  public void set(org.quickfix.field.TradeDate value) 
  { setField(value); } 
  public org.quickfix.field.TradeDate get(org.quickfix.field.TradeDate value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.TradeDate getTradeDate() throws FieldNotFound 
  { org.quickfix.field.TradeDate value = new org.quickfix.field.TradeDate(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.TradeDate field) 
  { return isSetField(field); } 
  public boolean isSetTradeDate() 
  { return isSetField(75); } 

  public void set(org.quickfix.field.TransactTime value) 
  { setField(value); } 
  public org.quickfix.field.TransactTime get(org.quickfix.field.TransactTime value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.TransactTime getTransactTime() throws FieldNotFound 
  { org.quickfix.field.TransactTime value = new org.quickfix.field.TransactTime(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.TransactTime field) 
  { return isSetField(field); } 
  public boolean isSetTransactTime() 
  { return isSetField(60); } 

  public void set(org.quickfix.field.AllocStatus value) 
  { setField(value); } 
  public org.quickfix.field.AllocStatus get(org.quickfix.field.AllocStatus value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AllocStatus getAllocStatus() throws FieldNotFound 
  { org.quickfix.field.AllocStatus value = new org.quickfix.field.AllocStatus(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AllocStatus field) 
  { return isSetField(field); } 
  public boolean isSetAllocStatus() 
  { return isSetField(87); } 

  public void set(org.quickfix.field.AllocRejCode value) 
  { setField(value); } 
  public org.quickfix.field.AllocRejCode get(org.quickfix.field.AllocRejCode value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.AllocRejCode getAllocRejCode() throws FieldNotFound 
  { org.quickfix.field.AllocRejCode value = new org.quickfix.field.AllocRejCode(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.AllocRejCode field) 
  { return isSetField(field); } 
  public boolean isSetAllocRejCode() 
  { return isSetField(88); } 

  public void set(org.quickfix.field.Text value) 
  { setField(value); } 
  public org.quickfix.field.Text get(org.quickfix.field.Text value) throws FieldNotFound 
  { getField(value); return value; } 
  public org.quickfix.field.Text getText() throws FieldNotFound 
  { org.quickfix.field.Text value = new org.quickfix.field.Text(); 
    getField(value); return value; } 
  public boolean isSet(org.quickfix.field.Text field) 
  { return isSetField(field); } 
  public boolean isSetText() 
  { return isSetField(58); } 
} 
